app/v2/controllers/V2MovementsController.scala

```diff
@@ -475,68 +470,103 @@ class V2MovementsControllerImpl @Inject() (
+                presentationError => Future.successful(Status(presentationError.code.statusCode)(Json.toJson(presentationError))),
+                {
+                  case (downloadUrl, size) if size <= config.smallMessageSizeLimit =>
+                    val messageTypeList =
+                      if (movementType == MovementType.Arrival) MessageType.messageTypesSentByArrivalTrader
+                      else MessageType.messageTypesSentByDepartureTrader
+                    (for {
+                      upscanFile <- upscanService.upscanGetFile(downloadUrl).asPresentation
+
+                      updateMovementResponse <- streamLargeMessage(eori, movementId, messageId, messageTypeList, upscanFile, movementType)
+
+                    } yield updateMovementResponse)
+                      .fold[Result](
+                        prentationError => {
+                          pushNotificationsService.postPpnsNotification(movementId, messageId, Json.toJson(prentationError))
+                          Ok
+                        },
+                        _ => {
+                          pushNotificationsService
+                            .postPpnsNotification(movementId, messageId, Json.toJson(HateoasMovementUpdateResponse(movementId, messageId, movementType)))
+                          Ok
+                        }
+                      )
+                  case (downloadUrl, size) if size > config.smallMessageSizeLimit =>
+                    (for {
+                      objectSummary <- objectStoreService.addMessage(downloadUrl, movementId, messageId).asPresentation
+                      uri = ObjectStoreResourceLocation(objectSummary.location.asUri)
+                      source      <- objectStoreService.getMessage(uri.stripOwner).asPresentation
+                      messageType <- xmlParsingService.extractMessageType(source, MessageType.values).asPresentation
+                      messageUpdate = MessageUpdate(_, Some(ObjectStoreURI(objectSummary.location.asUri)), Some(messageType))
+                      persist       = persistenceService.updateMessage(eori, movementType, movementId, messageId, _)
+                      _ <- validationService
+                        .validateLargeMessage(messageType, uri)
+                        .asPresentation
+                        .leftMap {
+                          err =>
+                            persist(messageUpdate(MessageStatus.Failed)).value
+                            err
+                        }
+                      _ <- persist(messageUpdate(MessageStatus.Processing)).asPresentation
+                      sendMessage <- routerService
+                        .sendLargeMessage(
+                          messageType,
+                          eori,
+                          movementId,
+                          messageId,
+                          ObjectStoreURI(objectSummary.location.asUri)
+                        )
+                        .asPresentation
```

Comment:
by Mistake Derek has removed below code from here. So please add that back again here

 ``` .leftMap {
  err =>
    persist(messageUpdate(MessageStatus.Failed)).value
    err
} ```